Zend/zend_compile.c

```diff
@@ -4837,21 +4833,32 @@ static zend_result zend_compile_func_sprintf(znode *result, zend_ast_list *args)
 			opline = zend_compile_rope_add(result, rope_elements++, &const_node);
 		}
 
-		if (*q == 's') {
-			/* Perform the cast of constant arrays when actually evaluating corresponding placeholder
-			 * for correct error reporting.
-			 */
-			if (elements[string_placeholder_count].op_type == IS_CONST) {
-				if (Z_TYPE(elements[string_placeholder_count].u.constant) == IS_ARRAY) {
-					zend_emit_op_tmp(&elements[string_placeholder_count], ZEND_CAST, &elements[string_placeholder_count], NULL)->extended_value = IS_STRING;
+		if (*q != '%') {
+			switch (*q) {
+			case 's':
+				/* Perform the cast of constants when actually evaluating the corresponding placeholder
+				 * for correct error reporting.
+				 */
+				if (elements[placeholder_count].op_type == IS_CONST) {
+					if (Z_TYPE(elements[placeholder_count].u.constant) == IS_ARRAY) {
+						zend_emit_op_tmp(&elements[placeholder_count], ZEND_CAST, &elements[placeholder_count], NULL)->extended_value = IS_STRING;
```

Comment:
Is this cast actually needed? `ZEND_ROPE_(INIT|ADD)` already handle non-strings.

Reply:
Oh, sorry. This is for consts.